# Workshops & Tutorials

## Financial and Legal Dimensions of Healthcare
*Organizers*: Barry Smith and William Hogan

The Ontology of Medically Related Social Entities (OMRSE) is being developed to provide a framework for modeling demographic data that is compatible with BFO and with Open Biomedical Ontologies (OBO) Foundry best practices. Recent developments include representations of organizations, roles, facilities, demographic data, enrollment in insurance plans, and data about socio-economic indicators. At the same time, a number of ontology groups have been working on BFO-conformant ontologies in the financial and legal realms, covering phenomena such as insurance or liability at a more general level. The goal of this workshop is to provide a forum in which the contributions of ontologies in these domains can be critically reviewed, and potential opportunities for collaboration identified.

*Time*: 1:30 pm – 4:45 pm, Monday, September 26, 2022   
*Presentations*: **[Google Drive](https://drive.google.com/drive/folders/1MTsP-hWXuFVJFR_Hch7KTUk43AgLln1J)**

#### Agenda
```txt
1:30 pm Mathias Brochhausen, University of Arkansas for Medical Sciences	 
Legal entities and Trauma care - the TIPTOE Perspective

2:15 pm Matt Diller, University of Florida	 
Representing Ownership and Access to Material Needs in OWL 2

3:00 pm BREAK	

3:15 pm Ryan Ricucci, US Customs and Border Patrol	
Healthcare on the Border: An Ontological View

4:00 pm Barry Smith, University at Buffalo	
The Ontology of Social Services 
```

## The 6th International Cells in Experimental Life Science Workshop, CELLS 2022
*Organizers*: Alexander D. Diehl and Yongqun He   
*Website*: [https://sites.google.com/view/cells-2022-workshop/home](https://sites.google.com/view/cells-2022-workshop/home)<br>
*YouTube*: [https://www.youtube.com/watch?v=s1ALmY3xxec](https://www.youtube.com/watch?v=s1ALmY3xxec)

The 6th International Cells in Experimental Life Science Workshop, CELLS 2022, will provide a venue for discussions of the application of biomedical ontologies to represent and analyze in vivo and in vitro cell- and cell line-related knowledge and data, including single cell RNA sequencing data and stem cell technologies.  Current high throughput methods such as single cell RNA sequencing and flow and mass cytometry are producing a large amount of data related to existing and novel cell types in health and disease. At the same time, experimental approaches such as microscopy, genomics, and metabolomics are expanding understanding of cellular functioning in relation to neighboring cells and the whole organism. Ontologies are being increasingly used as a tool for integrating and analyzing these diverse data types by projects such as HuBMAP, the Human Cell Atlas, and Brain Data Standards. The Cell Ontology (CL) and Cell Line Ontology (CLO) have long been established as reference ontologies in the OBO framework for representing cell type and cell line information, and additional ontologies such as the Gene Ontology, Protein Ontology, and the Ontology for Biomedical Investigation are also important for representing not only experimental data about cell types but also the methods used to produce that data. The workshop will cover the extension of the Cell Ontology (CL) and related ontologies for ontological representation of cell types based on new methodologies and experiments.

CELLS 2022 will take place as a half-day virtual workshop and will be free for all attendees.

Please visit the CELLS [website](https://sites.google.com/view/cells-2022-workshop/home) for submission details.

## FAIR ontology harmonization and TRUST data interoperability (FOHTI)
*Date and time*: Monday, September 26, 2022 (9am-12:30pm ET)<br>
*Organizers*: Asiyah Yu Lin, Gary Berg-Cross, Nomi Harris<br>
*Contact*: asiyah.lin@nih.gov;gbergcross@gmail.com<br>
*Website*: [FOHTI-22](https://github.com/ICBO-conference/icbo2022/blob/main/docs/FOHTI.md)

Many diverse ontologies and, increasingly, knowledge graphs (KGs) and other semantic resources have been developed across the biomedical and other domains. Although ontologies and associated KGs are often developed and used for specific needs there may be common knowledge across a specific domain as well as at a higher level, making some degree of knowledge and data alignment possible across many domains. To avoid organizations spending time and resources to model and represent that common knowledge, it is desirable to develop some consensus on a range of relevant semantic resources. Many types of these semantic resources exist along a semantic spectrum that includes structured vocabularies and high-level domain ontologies to support a wide range of use cases.

A relevant context for this work is the Findability, Accessibility, Interoperability, and Reuse (FAIR) principle and the Transparency, Responsibility, User Focus, Sustainability (TRUST) principle, which have been established and accepted by the global scientific community for digital objects. Can these principles also be applied to ontologies and other semantic resources which support ample opportunities for data discovery, trust, sharing, reuse and value of various datasets; as well as enabling wide access to dataset quality information? To accomplish all these goals at the data level, FAIR and TRUSTworthy ontology harmonization is an important first step. To do this we need to develop, improve, and disseminate community agreed upon best practices for harmonizing semantic resources at all levels.

This workshop aims to look for examples, technologies, and methodologies utilized for developing FAIR ontology and harmonization, as well as to discuss how harmonization across the semantic spectrum will facilitate data interoperability and the TRUST principle. In this workshop, we will discuss what has worked to create FAIR and TRUST principles that can help harmonize semantic resources, and what FAIR and TRUSTworthy domain-engaged semantic resource harmonization governance might look like.

## Workshop on Ontology Tools and Workflows
*Organizers*: James A. Overton, Charles Tapley Hoyt and Christopher J. Mungall<br>
*Website*: [ICBO 2022 Workshop on Ontology Tools and Workflows](https://docs.google.com/document/d/1zswfYOVI7YI46lg5kOuig5tAFOLnkC1d-rWYQHCC9SA/edit#)

Tools and workflows for ontology development and ontology use have grown increasingly powerful over the past decade, building on trends and tools from the larger open source software development community. ICBO 2017 included a workshop on tools and workflows for ontology editing, and this workshop ICBO 2022 provides an opportunity for tool developers and users to come together and discuss advances and changes of the past five years.

This year's hybrid in-person and remote workshop is divided into two parts:

OWL Beyond Java: OWLAPI has been the foundation of ontology tooling, especially in the OBO community. Editors such as Protege and automation tools such as ROBOT rely on OWLAPI to do the heavy lifting for their core operations. Recently a number of projects have emerged that can provide some of the OWLAPI's functionality with a different set of benefits and tradeoffs, such as: access from other languages, such as Python; alternative APIs; reduced memory usage and greater speed for certain operations; persisting OWL axioms to a database; and more. The first half of the workshop will include presentations on these various new tools and a group discussion of their applications, limitations, and benefits.

Convergence: Interoperability is a central goal for linked data technologies and the OBO community, but we face many obstacles and have limited resources. The theme of this session is convergence on: standards such as prefixes and CURIEs; term access and metadata; quality control rules and reporting; best practices for open source development; workflows such as imports and modules; and editing tools, and their integration into workflows. In our group discussion we will consider how to connect and coordinate development, without sacrificing the benefits of diversity in our open source community.

## 11th Vaccine and Drug Ontology Studies (VDOS-2022)
*Organizers*: Junguk Hur, Cui Tao and Yongqun He   
*Contact*: Cui Tao, cui.tao@uth.tmc.edu  
*Website*: [https://sites.google.com/view/vdosworkshop/vdos-2022](https://sites.google.com/view/vdosworkshop/vdos-2022)   
*YouTube*: [https://www.youtube.com/playlist?list=PLhzFEi0G-n-sdlkToWTdfqWLAqvcjGa6w](https://www.youtube.com/playlist?list=PLhzFEi0G-n-sdlkToWTdfqWLAqvcjGa6w)

 Drugs and vaccines have contributed to dramatic improvements in public health worldwide. Over the last decade, there have been efforts in the biomedical ontology community that represents various areas associated with drugs including vaccines that extend existing health and clinical terminology systems (e.g., SNOMED, RxNorm, NDF-RT, and MedDRA), vernacular medical terminologies, and their applications to research and clinical data. This workshop will provide a platform for discussing innovative solutions as well as the challenges in the development and application of biomedical ontologies to representing and analyzing drugs and vaccines, their administration, immune responses induced, adverse events, and similar topics. The workshop will cover two main areas: (i) ontology representation of drugs and vaccines, and (ii) applications of the ontologies in real-world situations – administration, adverse events, etc. Examples of biomedical subject matter in the scope of this workshop: drug components (e.g., drug active ingredients, vaccine antigens, and adjuvants), administration details (e.g., dosage, administration route, and frequency), gene immune responses and pathways, drug-drug or drug-food interactions, and adverse events. Both research and clinical subjects will be covered. We will also focus on computational methods used to study these, for example, literature mining of vaccine/drug-gene interaction networks, meta-analysis of host immune responses, and time event analysis of the pharmacological effects. This workshop is expected to support a deeper understanding of vaccine and drug mechanisms and effects using ontologies. More specific topics will be selected based on attendees’ submissions and interests.

## OBO Tutorial: Using and Reusing Ontologies
*Organizers*: James A. Overton, Rebecca Jackson, Chris Mungall, Nicole Vasilevsky, Nicolas Matentzoglu and Randi Vita   
*Website*: [https://oboacademy.github.io/obook/courses/icbo2022/](https://oboacademy.github.io/obook/courses/icbo2022/)

The Open Biological and Biomedical Ontologies (OBO) community includes hundreds of open source scientific ontology projects, committed to shared principles and practices for interoperability and FAIR data. An OBO tutorial has been a regular feature of ICBO for a decade, introducing new and experienced ontology users and developers to ontologies in general, and to current OBO tools and techniques specifically. While ICBO attracts many ontology experts, it also includes an audience of ontology beginners, and of ontology users looking to become ontology developers or to further refine their skills. Our OBO tutorial will help beginner and intermediate ontology users with a combination of theory and hands-on practice.

## Food Process Ontology Hackathon
*Organizers*: Damion Dooley, Tarini Naravane, Matthew Lange, Chen Yang and Hande Küçük McGinty
*Website*: [https://foodon.org/food-process-ontology-hackathon-icbo-2022/](https://foodon.org/food-process-ontology-hackathon-icbo-2022/)

A food processing discussion group within the Joint Food Ontology Workgroup has proposed a general OBO Foundry compatible food processing ontology and more specific recipe model which is currently focused on the combination of ingredients, devices, and recipe steps. The recipe model needs to be tested out on real-world recipe databases, to develop the kinds of RDF data structures and queries that are of practical use within food science, but which also have direct application in interoperability with other ontology-driven models of nutrient estimation, food related individual and population health, environmental impact of ingredients, and food supply chain traceability. This workshop aims to create RDF views of other recipe database content from the perspective of OBO Foundry ontologies, including FoodOn, ChEBI, and units of measure ontologies. Hacking can take on a number of challenges including Named Entity Recognition of recipe ingredients in free text recipes, to the representation of multi-component foods such as contained in food composition databases (e.g. world.openfoodfacts.org), to the transformation of semi-normalized recipe databases into pure food process ontology form. Additional work on expanding the recipe ontology representation beyond its “lab bench” purview into more nuanced cultural and historical representation of food knowledge is welcome too!

## LinkML Workshop/Tutorial
*Date and time*: Sunday, September 25, 2022 (9am-12pm ET)<br>  
*Organizers*: Christopher Mungall, Sierra Moxon, Mark Miller, Nomi Harris and Tim Putnam  
*Website*: [Google Doc](https://docs.google.com/document/d/1JWh7ddLQ74dBTY2QQFHBfW9XkJo9WDq6O33vI-wbQvc/edit#heading=h.um3x5zjmy6jk)   
*YouTube*: [https://youtu.be/1ukQ7yXdlWc](https://youtu.be/1ukQ7yXdlWc)

The [Linked data Modeling Language](https://linkml.io) is an object-oriented data modeling framework that aims to bring semantic web standards into a familiar modeling paradigm, simplifying the production of FAIR ontology-ready data. It can be used for schematizing a variety of kinds of data, ranging from simple flat checklist-style standards to complex interrelated normalized data utilizing polymorphism/inheritance. One major benefit to LinkML is that the framework provides not only the modeling components but also the software and tools needed to load, output, and validate data conforming to a LinkML model. 

This tutorial, aimed primarily at biocurators and data modelers, will be a hands-on exploration of the LinkML modeling framework including a discussion of its included tools that help the user work with data conforming to a LinkML model. We will guide the attendees through designing a new model that exercises the main LinkML modeling components, using LinkML tools to serialize and validate instances of the new model, and will discuss the many ways to author and maintain a LinkML model. 

Data integration is a major challenge in the life sciences, due to heterogeneity, complexity, the proliferation of ad-hoc formats and data structures, and poor compliance with FAIR guidelines. 
Often, the first step of integrating data from an external source is to figure out the similarities and differences in domain concepts across disciplines. Beyond this task, the technical implementation of the model influences how scientific concepts are related to a data source. An aggregator needs to understand how modeling choices are influenced by technical implementations in addition to scientific domains in order to bring resources together. Real barriers exist because we can’t map models to one another. Examples of groups attempting to solve these data integration problems are found in projects such as the Cancer Research Data Commons, the NCATS Data Translator, NMDC, and the Monarch Initiative. 

Ontologies and controlled vocabularies are a necessary part of the solution, providing a formal vocabulary of all the entities in a domain. However, ontologies alone are insufficient, as they are not intended for modeling the data itself. A wide variety of frameworks and methods for modeling data exist, such as JSON-Schema, UML, and relational database schemas, but these are typically tied to an underlying concrete representation and do not make data maximally interoperable. 

LinkML is a modeling framework that provides software and tools needed to load, dump, and validate data conforming to a LinkML model. 


*Prerequisites* 
This workshop is open to all interested participants, and listening through the hands-on aspects is encouraged with or without participating directly. 
For hands-on training pieces, basic familiarity with running commands (python scripts, bash commands) from the command line (in a terminal) will be assumed. Please complete _Lesson0_ (TBD) prior to the tutorial session. 
Please also create a GitHub account if you do not already have one. The OBO Semantic Engineering Training has a nice tutorial on getting started with GitHub. 
We will also provide a docker container. If docker is preferred, docker or docker desktop should be installed. 

*Learning Objectives* 
Learn how to author a new LinkML model that exercises some of the main modeling components. 
Generate JSONSchema, SQL-DDL, Python, and Java classes from the new model. 
Survey several pre-built ways of authoring models. 
Learn about DataHarmonizer; a standardized spreadsheet editor and validator that can be used to curate data conforming to a custom LinkML model. 

## First Workshop on the Role of Ontologies in Biomedical AI (ROBI)
*Organizers*: Robert Hoehndorf and George Gkoutos    
*Contact*: Robert Hoehndorf, robert.hoehndorf@kaust.edu.sa   
*Website*: [https://bio-ontology-research-group.github.io/icbo-2022-ml-workshop/](https://bio-ontology-research-group.github.io/icbo-2022-ml-workshop/)   

Ontologies and Linked Data resources are pervasive throughout the life sciences where hundreds of ontologies are used in almost every major biological database for the formal characterization of data and the integration of knowledge. Methods that allow incorporating the knowledge in ontologies in biomedical data analysis and particular machine learning are only now being developed. The aim of the workshop is to explore the interface between domain knowledge in biomedical ontologies and machine learning from the perspective of ontologies, their structure, and underlying conceptualization. We will discuss how ontologies can contribute to build more comprehensive AI models that utilize domain knowledge, provide explanations understandable to domain experts, and identify consistency with scientific knowledge.

